test/src/org/labkey/test/tests/mobileappstudy/TokenValidationTest.java

```diff
@@ -216,6 +217,35 @@ public void testResolveEnrollmentToken()
         assertEquals(STUDY_NAME03, resolveCmd.getStudyId());
     }
 
+    @Test
+    public void testMyStudiesCoordinatorRole()
+    {
+        goToProjectHome(PROJECT_NAME01);
+        SetupPage setupPage = new SetupPage(this);
+        TokenBatchesWebPart batchesWebPart = setupPage.getTokenBatchesWebPart();
+
+        // Test for Administrator
+        assertTrue(batchesWebPart.isNewBatchPresent());
+        assertTrue(batchesWebPart.isNewBatchEnabled());
+        setupPage.validateSubmitButtonDisabled();  // Submit button should be present
+
+        // Test for Reader
+        impersonateRole("Reader");
+        assertTrue(batchesWebPart.isNewBatchPresent());
+        assertFalse(batchesWebPart.isNewBatchEnabled());
+        assertFalse(setupPage.isSubmitButtonVisible());  // Submit button should NOT be present
+        stopImpersonating(false);
+
+        // Test for MyStudies Coordinator
+        impersonateRoles("Reader", "MyStudies Coordinator");
+        assertTrue(batchesWebPart.isNewBatchPresent());
+        assertTrue(batchesWebPart.isNewBatchEnabled());  // Should be able to create a new batch
+        assertFalse(setupPage.isSubmitButtonVisible());  // Submit button should NOT be present
+        TokenBatchPopup tokenBatchPopup = batchesWebPart.openNewBatchPopup();
+        tokenBatchPopup.createNewBatch("100");
```

Comment:
Should we validate that the tokens are actually created? The UI might show the buttons but the server might reject the request.
```suggestion
        TokenListPage tokenListPage = tokenBatchPopup.createNewBatch("100");
        assertEquals("Wrong number of tokens generated for MyStudies Coordinator", 100, tokenListPage.getNumTokens());
```